R/pkg/R/DataFrame.R

```diff
@@ -1217,29 +1217,38 @@ setMethod("[[", signature(x = "DataFrame", i = "numericOrcharacter"),
 
 #' @rdname subset
 #' @name [
-setMethod("[", signature(x = "DataFrame", i = "missing"),
-          function(x, i, j, ...) {
-            if (is.numeric(j)) {
-              cols <- columns(x)
-              j <- cols[j]
-            }
-            if (length(j) > 1) {
-              j <- as.list(j)
+setMethod("[", signature(x = "DataFrame"),
+          function(x, i, j, ..., drop=T) {
+            # Perform filtering first if needed
+            filtered <- if (missing(i)) {
+              x
+            } else {
+              if (class(i) != "Column") {
+                stop(paste0("Expressions other than filtering predicates are not supported ",
+                      "in the first parameter of subsetting operator [."))
```

Comment:
suggest `extract operator [ or subset()`, as [ is referred to as 'extract' [here](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Extract.html), and subset() calls also go here